File: app/src/main/java/com/gooner10/popularmoviesapp/Activity/moviemain/MovieFragment.java
package com.gooner10.popularmoviesapp.Activity.moviemain;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.gooner10.popularmoviesapp.Activity.data.MovieData;
import com.gooner10.popularmoviesapp.R;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Movie Fragment to display the Movie List in GridLayout
 */

public class MovieFragment extends Fragment implements MovieContract.View {
    private final String LOG_TAG = "MovieFragment";
    private MovieFragmentAdapter movieFragmentAdapter;
    MovieContract.UserActionsListener movieListPresenter = new MovieListPresenter(this);

    @BindView(R.id.recyclerViewMovie)
    RecyclerView movierecyclerview;

    public MovieFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        movieListPresenter.loadData();

        View view = inflater.inflate(R.layout.fragment_movie, container, false);

        // Bind all of the view
        ButterKnife.bind(this, view);

        movierecyclerview.setLayoutManager(new GridLayoutManager(getActivity(), 2));
        movieFragmentAdapter = new MovieFragmentAdapter(getActivity(), new ArrayList<MovieData>());
        movierecyclerview.setAdapter(movieFragmentAdapter);

        return view;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
    }

    @Override
    public void displayMovieList(ArrayList<MovieData> movieDataList) {
        movieFragmentAdapter.setData(movieDataList);
        Log.i(LOG_TAG, "displayMovieList");
    }

}

File: feature/moviehome/src/main/java/com/gooner10/moviehome/MovieFragment.kt
package com.gooner10.moviehome

import android.content.Context
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.GridLayoutManager
import com.gooner10.data.model.MovieItem
import com.gooner10.moviehome.databinding.FragmentMovieBinding
//import hugo.weaving.DebugLog
//import kotlinx.android.synthetic.main.fragment_movie.view.*
import java.util.*

/**
 * Movie Fragment to display the Movie List in GridLayout
 */

class MovieFragment : Fragment(), MovieContract.View {
    companion object {
        private val TAG = "MovieFragment"
    }

    private var movieFragmentAdapter: MovieFragmentAdapter? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val binding = FragmentMovieBinding.inflate(inflater, container, false)
//        val view = inflater.inflate(R.layout.fragment_movie, container, false)

        binding.recyclerViewMovie.layoutManager = GridLayoutManager(requireActivity(), 2)
        movieFragmentAdapter = MovieFragmentAdapter(requireActivity() as Context, ArrayList())
        binding.recyclerViewMovie.adapter = movieFragmentAdapter
        Log.i(TAG, "onCreateView movieFragmentAdapter: " + movieFragmentAdapter!!)
        val movieViewModel = ViewModelProviders.of(requireActivity()).get(MovieViewModel::class.java)
        movieViewModel.getMovieDataList().observe(requireActivity(), Observer { movieDataList -> displayMovieList(movieDataList) })
        return binding.root
    }

    //@DebugLog
    override fun displayMovieList(movieDataList: List<MovieItem>) {
        Log.i(TAG, "displayMovieList movieFragmentAdapter: " + movieFragmentAdapter!!)
        movieFragmentAdapter!!.setData(movieDataList)
    }

}
